vendor/engines/split_accounts/lib/split_accounts/engine.rb

```diff
@@ -1,22 +1,32 @@
 module SplitAccounts
   class Engine < Rails::Engine
 
-    # Include modules in main rails app
     config.to_prepare do
-      Account.send :include, SplitAccounts::AccountExtensions
+      if SettingsHelper.feature_on?(:split_accounts)
+        # Include modules in main rails app
+        Account.send :include, SplitAccounts::AccountExtension
+
+        # Concat class variables in main rails app
+        Account.config.account_types << "SplitAccounts::SplitAccount"
+
+        # Add views to view hooks in main rails app
+        ViewHook.add_hook "facility_accounts.show", "after_end_of_form", "split_accounts/facility_accounts/show_splits"
+      end
     end
 
     # Include migrations in main rails app
     # https://blog.pivotal.io/labs/labs/leave-your-migrations-in-your-rails-engines
     initializer :append_migrations do |app|
-      unless app.root.to_s.match root.to_s
+      unless app.root.to_s.match(root.to_s)
         app.config.paths["db/migrate"] += config.paths["db/migrate"].expanded
       end
     end
 
     # Include factories in main rails app
     initializer "model_core.factories", after: "factory_girl.set_factory_paths" do
-      FactoryGirl.definition_file_paths << File.expand_path('../../../spec/factories', __FILE__) if defined?(FactoryGirl)
+      if defined?(FactoryGirl) && SettingsHelper.feature_on?(:split_accounts)
```

Comment:
Where were you running into problems with FactoryGirl not being defined. I would expect the `after: "factory_girl.set_factory_paths"` to ensure that FG is always there.